vochain/proof.go

```diff
@@ -70,15 +71,26 @@ func VerifyProofOffChainTree(process *models.Process, proof *models.Proof,
 	}
 	switch proof.Payload.(type) {
 	case *models.Proof_Graviton:
-		p := proof.GetGraviton()
-		if p == nil {
-			return false, nil, fmt.Errorf("graviton proof is empty")
-		}
-		valid, err := gravitontree.CheckProof(key, []byte{}, censusRoot, p.Siblings)
-		return valid, big.NewInt(1), err
+		return false, nil, fmt.Errorf("graviton proof no longer supported")
 	case *models.Proof_Iden3:
 		// NOT IMPLEMENTED
 		return false, nil, fmt.Errorf("iden3 proof not implemented")
+	case *models.Proof_Arbo:
+		p := proof.GetArbo()
+		if p == nil {
+			return false, nil, fmt.Errorf("arbo proof is empty")
+		}
+		var hashFunc arbo.HashFunction = arbo.HashFunctionBlake2b
+		switch p.Type {
+		case models.ProofArbo_BLAKE2B:
+			hashFunc = arbo.HashFunctionBlake2b
+		case models.ProofArbo_POSEIDON:
+			hashFunc = arbo.HashFunctionPoseidon
+		default:
+			return false, nil, fmt.Errorf("not recognized ProofArbo type: %s", p.Type)
+		}
+		valid, err := tree.VerifyProof(hashFunc, key, []byte{}, p.Siblings, censusRoot)
+		return valid, big.NewInt(1), err
```

Comment:
nit: do you need to return a new big int every time? if the caller does not modify the value, maybe you could use a global big int that's always 1.

Reply:
Agree, this was in this way previous to this PR, but agree to use a global big int. Updated it, and also for the `big.Int(0)` that is used in a couple of places in the same file.